doc/pages/command-parsing.asciidoc

```diff
@@ -49,7 +50,34 @@ of `(`, `[`, `{` and `<`), it is parsed as a balanced string whose closing
 delimiter matches that of its opening delimiter (respectively, `)`, `]`,
 `}`, and `>`).
 
-Characters may be escaped in the same manner as those for quoted strings.
+There is no way to escape the opening and closing characters, even if they
+are nested inside some other kind of string. For example, this will **not**
+work, because the `{` in the map command interferes with the delimiters for the
+hook's command block:
+
+----
+# DOES NOT WORK
+hook global WinSetOption filetype=latex %{
+    map window user b 'o\begin{'
```

Comment:
Looks fine.

It is a bit weird to place the more complex examples before the simple ones, not sure how to best fix that.

Reply:
agreed, I'd rather add simple not working examples in the Examples sections such as:

```
- %{ foo\{ } is an unterminated balanced string wheras %[ bar\{ ] is terminated
```

I would not mention the comment trick in here as I do not think its something that needs to be encouraged by the docs (I am not against using that trick, but putting it in the docs feels like an endorsement)

Reply:
I actually forgot there was a "Balanced String Examples" section right there. I've updated the PR to include examples more in line (in style and location) with the existing ones.